### PR TITLE
Multipage evals with HAR support

### PR DESCRIPTION
**Pull request by KhoomeiK**
- supports creation, fusion, and serving HARs in eval
- added 3 HAR examples
- removes fetch_id vs goal exclusivity. more work needed here

### CODE REVIEW

File: bananalyzer/runner/website_responder.py [hunk 1]
```diff
@@ -45,6 +45,8 @@ def get_url(self, example: Example) -> str:
 def get_website_responder(example: Example) -> WebsiteResponder:
     if example.source == "mhtml":
         return MHTMLWebsiteResponder(get_examples_path())
+    elif example.source == "har":
+        return HostedWebsiteResponder()  # HTTP requests are routed to the HAR file
```
Comment (awtkns) [inline bananalyzer/runner/website_responder.py:49]:
```suggestion
        return HostedWebsiteResponder()
```

File: tests/test_example_eval.py
Comment (awtkns) [inline tests/test_example_eval.py:1]: Add an e2e test

File: bananalyzer/runner/runner.py [hunk 2]
```diff
@@ -84,8 +84,9 @@ async def page():
             ignore_https_errors=True,
             user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36",
         )
+        {f"await context.route_from_har('{test_content.example.har_file_path}', not_found='abort', update=False)" if test_content.example.source == "har" else ""}
         page = await context.new_page()
-        await stealth_async(page)
+        {"await stealth_async(page)" if test_content.example.source != "har" else ""}
```
Comment (awtkns) [inline bananalyzer/runner/runner.py:89]: remove

File: static/examples.json
Comment (asim-shrestha) [inline static/examples.json:1]: Also need to update the test set examples

File: bananalyzer/data/schemas.py [hunk 3]
```diff
@@ -132,6 +136,32 @@ def get_static_url(self) -> str:
...
+                    else:
+                        raise ValueError(
+                            f"Could not find HAR file at {har_path}. Please ensure it has been downloaded from S3 to the correct location."
+                        )
+                else:
+                    return get_examples_path() / self.resource_path
+            else:
+                raise ValueError("This example does not have a resource path")
+        else:
+            raise ValueError("This example is not a HAR file")
```
Comment (awtkns) [inline bananalyzer/data/schemas.py:163]: invert all these conditions

if not "har":
  raise
  
  
  
  etc..

File: bananalyzer/__main__.py [hunk 4]
```diff
@@ -323,13 +324,35 @@ def main() -> int:
...
 
     for example in examples:
-        if example.mhtml_url is not None:
-            mhtml_path = get_examples_path() / example.id / "index.mhtml"
-            if not mhtml_path.exists():
-                mhtml_str = download_mhtml(example.mhtml_url)
-                mhtml_path.parent.mkdir(parents=True, exist_ok=False)
-                with open(mhtml_path, "w") as file:
-                    file.write(mhtml_str)
+        if example.resource_path is not None:
```
Comment (awtkns) [inline bananalyzer/__main__.py:327]: invert

**Review by awtkns — APPROVED** [review]

File: bananalyzer/runner/runner.py (hunk 2 above)
Comment (asim-shrestha) [inline bananalyzer/runner/runner.py:89]: Delete stealth async

File: static/examples.json
Comment (KhoomeiK) [reply static/examples.json:1]: none have `mhtml_url`

File: bananalyzer/__main__.py [hunk 5]
```diff
@@ -323,13 +324,35 @@ def main() -> int:
...
+                har_path = get_examples_path() / har_subpath
+                if not os.path.exists(har_path):
+                    har_dir_path = os.path.dirname(har_path)
+                    download_har(har_dir_path, example.resource_path)
+            elif example.resource_path.endswith("/index.har"):
+                har_subpath = "/".join(parts[-2:])
+                har_path = get_examples_path() / har_subpath
+                if not os.path.exists(har_path):
+                    raise ValueError(
+                        f"Could not find HAR file at {har_path}. Please run `bananalyze --download` to download all example files. No S3 path is provided for this example."
```
Comment (awtkns) [inline bananalyzer/__main__.py:350]: invert

File: bananalyzer/__main__.py [hunk 6]
```diff
@@ -323,13 +324,35 @@ def main() -> int:
...
+                    )
+            else:
+                raise ValueError(
+                    f"Could not find HAR resource at {example.resource_path}. Please ensure the example provides either the path to an index.har in a local bananalyzer examples subdirectory, or an S3 URL to a tar.gz of a HAR directory."
+                )
```
Comment (awtkns) [inline bananalyzer/__main__.py:355]: invert

File: tests/test_examples.py [hunk 7]
```diff
@@ -68,6 +71,62 @@ def test_download_examples() -> None:
...
+
+@pytest.mark.parametrize(
+    "contents, should_write",
+    [
+        ([{"name": "index.har", "data": b"dummy har content"}], True),
+        ([{"name": "largefile.bin", "data": b"x" * 10**6}], True),
+        ([], False),
+    ],
+)
+def test_download_har(mocker, mock_s3_client, contents, should_write):
```
Comment (awtkns) [inline tests/test_examples.py:100]: dont parameterize the should write, split it up here. it is an antipattern if you need if else assertions in your tests

Comment (awtkns) [reply tests/test_examples.py:100]: lets also add a with pytest.raises(....) test for an exception case

File: tests/test_example_eval.py [hunk 8]
```diff
@@ -201,12 +201,57 @@ def test_fetch_with_non_dictionary_goal() -> None:
...
+        ),
+        # Test empty resource path
+        (
+            {"resource_path": "", "source": "har"},
+            ValueError,
+            None,
+        ),
+    ],
+)
+def test_har_file_path(overrides, expected_exception, expected_path, mocker):
```
Comment (awtkns) [inline tests/test_example_eval.py:240]: split up the raises tests into thier own tests

File: bananalyzer/__main__.py (hunk 6 above)
Comment (KhoomeiK) [reply bananalyzer/__main__.py:355]: would result in conditional duplication

**Pull request merged**